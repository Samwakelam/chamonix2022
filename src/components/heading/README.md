# Heading

A basic HTML heading element, can be set to H1-H6 through the `size` prop. Content is passed in via `children` prop.

## Import

```jsx
import { Heading } from '@gymshark/design-system';
```

# Reference

## Props

| Prop       | Required | Type        | Description                                                                              |
| ---------- | -------- | ----------- | ---------------------------------------------------------------------------------------- |
| `size`     | `true`   | `ENUM`      | Import `HeadingSizes` for ENUM provider, changes font size and rendered HTML Heading tag |
| `children` | `true`   | `ReactNode` | Valid react children element                                                             |
