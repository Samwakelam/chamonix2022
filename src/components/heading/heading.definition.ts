export enum HeadingSizes {
    BANNER = 'banner',
    H1 = 'h1',
    H2 = 'h2',
    H3 = 'h3',
    H4 = 'h4',
    H5 = 'h5',
    H6 = 'h6',
}

type SizeObject = {
    mobile: number;
    tablet: number;
    desktop: number;
};

type SingleHeading = {
    element: HeadingSizes;
    fontSize: SizeObject;
    lineHeight: SizeObject;
};

export interface HeadingProps extends React.HTMLAttributes<HTMLHeadingElement> {
    children: React.ReactNode;
    size: HeadingSizes;
}

export type HeadingSizesShape = {
    [key: string]: SingleHeading;
};

export type HeadingStyledProps = {
    configuration: SingleHeading;
};
