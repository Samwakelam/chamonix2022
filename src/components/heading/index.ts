export { Heading } from './heading.component';
export { HeadingSizes } from './heading.definition';
