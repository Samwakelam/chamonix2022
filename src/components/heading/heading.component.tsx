import React from 'react';

import { HeadingProps, HeadingSizes } from './heading.definition';
import { headingSizes } from './heading.shape';

import * as S from './heading.styles';

export const Heading = ({ children, size = HeadingSizes.BANNER, ...props }: HeadingProps): JSX.Element => (
    <S.Heading configuration={headingSizes[size]} as={headingSizes[size].element} {...props}>
        {children}
    </S.Heading>
);
