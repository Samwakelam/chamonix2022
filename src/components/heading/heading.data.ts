import { HeadingProps, HeadingSizes } from './heading.definition';

export const HeadingData: HeadingProps = {
    children: 'WELCOME TO GYMSHARK',
    size: HeadingSizes.BANNER,
};
