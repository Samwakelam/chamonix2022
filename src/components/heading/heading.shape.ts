import { HeadingSizes, HeadingSizesShape } from './heading.definition';

export const headingSizes: HeadingSizesShape = {
    banner: {
        element: HeadingSizes.H1,
        fontSize: {
            mobile: 3.5,
            tablet: 4,
            desktop: 4.5,
        },
        lineHeight: {
            mobile: 4,
            tablet: 4.5,
            desktop: 5,
        },
    },
    h1: {
        element: HeadingSizes.H1,
        fontSize: {
            mobile: 2.5,
            tablet: 3,
            desktop: 3.2,
        },
        lineHeight: {
            mobile: 3,
            tablet: 3.5,
            desktop: 4,
        },
    },
    h2: {
        element: HeadingSizes.H2,
        fontSize: {
            mobile: 2,
            tablet: 2.5,
            desktop: 2.5,
        },
        lineHeight: {
            mobile: 2.5,
            tablet: 3,
            desktop: 3,
        },
    },
    h3: {
        element: HeadingSizes.H3,
        fontSize: {
            mobile: 1.8,
            tablet: 2,
            desktop: 2,
        },
        lineHeight: {
            mobile: 2.3,
            tablet: 2.5,
            desktop: 2.5,
        },
    },
    h4: {
        element: HeadingSizes.H4,
        fontSize: {
            mobile: 1.6,
            tablet: 1.6,
            desktop: 1.6,
        },
        lineHeight: {
            mobile: 2.1,
            tablet: 2.1,
            desktop: 2.1,
        },
    },
    h5: {
        element: HeadingSizes.H5,
        fontSize: {
            mobile: 1.4,
            tablet: 1.4,
            desktop: 1.4,
        },
        lineHeight: {
            mobile: 1.9,
            tablet: 1.9,
            desktop: 1.9,
        },
    },
    h6: {
        element: HeadingSizes.H6,
        fontSize: {
            mobile: 1.2,
            tablet: 1.2,
            desktop: 1.3,
        },
        lineHeight: {
            mobile: 1.7,
            tablet: 1.7,
            desktop: 1.8,
        },
    },
};
