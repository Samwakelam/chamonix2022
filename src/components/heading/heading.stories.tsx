import React from 'react';

import { Meta, Story } from '@storybook/react';
import '@storybook/addon-console';

import { Heading as HeadingComponent } from './heading.component';
import { HeadingProps, HeadingSizes } from './heading.definition';
import { HeadingData } from './heading.data';

export default {
    component: HeadingComponent,
    title: 'Atoms/Typography/Heading',
} as Meta;

const Heading: Story<HeadingProps> = ({ children, ...args }: HeadingProps) => (
    <HeadingComponent {...args}>{children}</HeadingComponent>
);

export const Banner = Heading.bind({});

Banner.args = {
    children: HeadingData.children,
    size: HeadingSizes.BANNER,
};

export const H1 = Heading.bind({});

H1.args = {
    children: HeadingData.children,
    size: HeadingSizes.H1,
};

export const H2 = Heading.bind({});

H2.args = {
    children: HeadingData.children,
    size: HeadingSizes.H2,
};

export const H3 = Heading.bind({});

H3.args = {
    children: HeadingData.children,
    size: HeadingSizes.H3,
};

export const H4 = Heading.bind({});

H4.args = {
    children: HeadingData.children,
    size: HeadingSizes.H4,
};

export const H5 = Heading.bind({});

H5.args = {
    children: HeadingData.children,
    size: HeadingSizes.H5,
};

export const H6 = Heading.bind({});

H6.args = {
    children: HeadingData.children,
    size: HeadingSizes.H6,
};
