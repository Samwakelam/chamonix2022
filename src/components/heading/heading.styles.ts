import styled, { css } from 'styled-components';
import { theme } from '../../theme';
import { HeadingStyledProps } from './heading.definition';

const sizeStyles = ({ fontSize, lineHeight }: { fontSize: number; lineHeight: number }) => css`
    font-size: ${fontSize}rem;
    line-height: ${lineHeight}rem;
`;

export const Heading = styled.h1<HeadingStyledProps>`
    font-family: ${theme.f.lato};
    font-weight: 700;
    text-transform: uppercase;

    /* ${({ configuration }) =>
        sizeStyles({ fontSize: configuration.fontSize.mobile, lineHeight: configuration.lineHeight.mobile })}

    @media ${device.tablet} {
        ${({ configuration }) =>
        sizeStyles({ fontSize: configuration.fontSize.tablet, lineHeight: configuration.lineHeight.tablet })}
    }

    @media ${device.desktopS} {
        ${({ configuration }) =>
        sizeStyles({ fontSize: configuration.fontSize.desktop, lineHeight: configuration.lineHeight.desktop })}
    } */
`;
