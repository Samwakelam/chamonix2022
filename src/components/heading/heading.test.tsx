import React from 'react';
import { render } from '@testing-library/react';
import { Heading } from './heading.component';
import { HeadingData } from './heading.data';

// describe('Heading Component', () => {
//     const { container } = render(<Heading {...HeadingData} />);

//     it('should match snapshot correctly', () => {
//         expect(container.firstChild).toMatchSnapshot();
//     });
// });
