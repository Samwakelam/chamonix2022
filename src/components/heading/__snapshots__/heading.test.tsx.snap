// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should match snapshot correctly 1`] = `
<h1
  class="sc-bdfBQB qtJAJ"
>
  WELCOME TO GYMSHARK
</h1>
`;
